src/data/chats/chat.rs

```diff
@@ -268,51 +276,60 @@ impl Chat {
             content: prompt.clone(),
         });
 
-        self.last_used_file_id = Some(loaded_file.id.clone());
-
         self.messages.push(ChatMessage {
             id: next_id + 1,
             role: Role::Assistant,
-            username: Some(loaded_file.name.clone()),
+            username: Some(wanted_file.name.clone()),
             content: "".to_string(),
         });
 
-        let store_chat_tx = self.messages_update_sender.clone();
-        backend.command_sender.send(cmd).unwrap();
         self.is_streaming = true;
-        thread::spawn(move || loop {
-            if let Ok(response) = rx.recv() {
-                match response {
-                    Ok(ChatResponse::ChatResponseChunk(data)) => {
-                        let mut is_done = false;
-
-                        let _ = store_chat_tx.send(ChatTokenArrivalAction::AppendDelta(
-                            data.choices[0].delta.content.clone(),
-                        ));
-
-                        if let Some(_reason) = &data.choices[0].finish_reason {
-                            is_done = true;
-                            let _ = store_chat_tx.send(ChatTokenArrivalAction::StreamingDone);
-                        }
 
-                        SignalToUI::set_ui_signal();
-                        if is_done {
+        let store_chat_tx = self.messages_update_sender.clone();
+        let wanted_file = wanted_file.clone();
+        let command_sender = backend.command_sender.clone();
+        thread::spawn(move || {
+            if let Err(err) = model_loader.load_blocking(wanted_file.id, command_sender.clone()) {
+                eprintln!("Error loading model: {}", err);
```

Comment:
Future idea: Eventually, all `eprintln` we have around in the codebase could trigger a `Cx::post_action(ErrorNotification(...))` with the latest updates in makepad, to get feedback in the UI.